drivers/sensor/max17055/max17055.c

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/i2c.h>
+#include <drivers/sensor.h>
+
+#include "max17055.h"
+
+#define DT_DRV_COMPAT maxim_max17055
+
+static int max17055_reg_read(struct max17055_data *max17055, int reg_addr,
+			     s16_t *val)
+{
+	u8_t i2c_data[2];
+	int status;
+
+	status = i2c_burst_read(max17055->i2c, DT_INST_REG_ADDR(0), reg_addr,
+				i2c_data, 2);
+	if (status < 0) {
+		LOG_ERR("Unable to read register");
+		return -EIO;
+	}
+
+	*val = i2c_data[1] << 8 | i2c_data[0];
+
+	return 0;
+}
+
+static int capacity_to_ma(unsigned int rsense_mohms, s16_t val)
+{
+	int lsb_units, rem;
+
+	/* Get units for the LSB in uA */
+	lsb_units = 5 * 1000 / rsense_mohms;
+	/* Get remaining capacity in uA */
+	rem = val * lsb_units;
+
+	return rem;
+}
+
+static void set_millis(struct sensor_value *val, int val_millis)
+{
+	val->val1 = val_millis / 1000;
+	val->val2 = val_millis % 1000 * 1000;
+}
+
+/**
+ * @brief sensor value get
+ *
+ * @return -ENOTSUP for unsupported channels
+ */
+static int max17055_channel_get(struct device *dev, enum sensor_channel chan,
```

Comment:
Wouldn't `channel_get_sensor_val` be a more appropriate name for this function ? The current name suggests that we are getting the channel (like dma channel) which is incorrect..